2020-01-Winter/3_stacks_queues/README.md

```diff
@@ -116,21 +157,75 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
+### 2. Recent Counter
 
-Source: TODO :bug:
+Source: https://leetcode.com/problems/number-of-recent-calls/solution/
 
 #### Naive/Simple Solution
-
-TODO :bug:
-
-#### Optimal Solution
-
-TODO :bug:
+**Time Complexity**: O(n) where N is the number of times `ping()` is called.
+
+**Space Complexity:** O(n) where N is the size of the window. In this case, 3000.
+
+Java Solution:
+```java
+class RecentCounter{
+    //This is one way of making a FIFO Queue in Java
+    Queue<Integer> queue = new LinkedList<>();
+
+    //The ping function.
+    //Calling this function requires passing in the time t in milliseconds
+    public int ping(int t){
+        //First, we eneueue the new time into the end of the queue
+        queue.add(t);
+
+        //Now we loop through the queue and dequeue from the front
+        //any pings that happened more that 3000 seconds ago
+        for(int i = 0; i < queue.size(); i++){
+            if(queue.peek() < t-3000){
+                queue.remove();
+            }
+        }
+
+        //Since the problem statement wants the number of pings in
+        //the previous 3000 milliseconds, we can now simply return
+        //the number of elements in the queue
+        return queue.size();
+    }
+}
+```
+
+Python Solution:
+```python
+class RecentCounter:
+    #This queue is a Python list
+    #In this problem, we treat it like a FIFO Queue
+    queue = []
```

Comment:
This is fine for a single example, but the variable `queue` is being treated as static to the class instead of specific to the instance. If I try to add new examples / class instances, the contents of the variable are retained.
```suggestion
    def __init__(self):
        self.queue = []
```